// sails-service-location/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

exports.default = function (type, config) {
  if (location[type.toLowerCase()] instanceof Function) {
    return new location[type.toLowerCase()](config);
  } else {
    throw new Error('Unrecognized type -> ' + type);
  }
};

var _AgolLocation = require('./AgolLocation');

var _AgolLocation2 = _interopRequireDefault(_AgolLocation);

var _DataScienceToolkitLocation = require('./DataScienceToolkitLocation');

var _DataScienceToolkitLocation2 = _interopRequireDefault(_DataScienceToolkitLocation);

var _FreeGeoIpLocation = require('./FreeGeoIpLocation');

var _FreeGeoIpLocation2 = _interopRequireDefault(_FreeGeoIpLocation);

var _GeoCodioLocation = require('./GeoCodioLocation');

var _GeoCodioLocation2 = _interopRequireDefault(_GeoCodioLocation);

var _GoogleLocation = require('./GoogleLocation');

var _GoogleLocation2 = _interopRequireDefault(_GoogleLocation);

var _HereLocation = require('./HereLocation');

var _HereLocation2 = _interopRequireDefault(_HereLocation);

var _MapQuestLocation = require('./MapQuestLocation');

var _MapQuestLocation2 = _interopRequireDefault(_MapQuestLocation);

var _NominatimMapQuestLocation = require('./NominatimMapQuestLocation');

var _NominatimMapQuestLocation2 = _interopRequireDefault(_NominatimMapQuestLocation);

var _OpenCageLocation = require('./OpenCageLocation');

var _OpenCageLocation2 = _interopRequireDefault(_OpenCageLocation);

var _OpenMapQuestLocation = require('./OpenMapQuestLocation');

var _OpenMapQuestLocation2 = _interopRequireDefault(_OpenMapQuestLocation);

var _OpenStreetMapLocation = require('./OpenStreetMapLocation');

var _OpenStreetMapLocation2 = _interopRequireDefault(_OpenStreetMapLocation);

var _SmartyStreetsLocation = require('./SmartyStreetsLocation');

var _SmartyStreetsLocation2 = _interopRequireDefault(_SmartyStreetsLocation);

var _TeleportLocation = require('./TeleportLocation');

var _TeleportLocation2 = _interopRequireDefault(_TeleportLocation);

var _TomTomLocation = require('./TomTomLocation');

var _TomTomLocation2 = _interopRequireDefault(_TomTomLocation);

var _YandexLocation = require('./YandexLocation');

var _YandexLocation2 = _interopRequireDefault(_YandexLocation);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var location = {
  agol: _AgolLocation2.default,
  datasciencetoolkit: _DataScienceToolkitLocation2.default,
  freegeoip: _FreeGeoIpLocation2.default,
  geocodio: _GeoCodioLocation2.default,
  google: _GoogleLocation2.default,
  here: _HereLocation2.default,
  mapquest: _MapQuestLocation2.default,
  nominatimmapquest: _NominatimMapQuestLocation2.default,
  opencage: _OpenCageLocation2.default,
  openmapquest: _OpenMapQuestLocation2.default,
  openstreetmap: _OpenStreetMapLocation2.default,
  smartystreets: _SmartyStreetsLocation2.default,
  teleport: _TeleportLocation2.default,
  tomtom: _TomTomLocation2.default,
  yandex: _YandexLocation2.default
};

/**
 * Create specified location instance
 * @param {String} type
 * @param {Object} [config]
 * @returns {*}
 */
module.exports = exports['default'];
